src/transaction/TransactionController.ts

```diff
@@ -1278,23 +1316,21 @@ export class TransactionController extends BaseController<
 
         return [meta, true];
       case TransactionStatus.submitted:
+        if (pollingAttempts === maxPollingAttempts) {
+          const error: Error = new Error(
+            'Transaction failed. The transaction was dropped or replaced by a new one',
+          );
+          this.failTransaction(meta, error);
+          return [meta, false];
+        }
+
         const txObj = await query(this.ethQuery, 'getTransactionByHash', [
           transactionHash,
         ]);
 
-        if (!txObj) {
-          const receiptShowsFailedStatus = await this.checkTxReceiptStatusIsFailed(
-            transactionHash,
-          );
-
-          // Case the txObj is evaluated as false, a second check will
-          // determine if the tx failed or it is pending or confirmed
-          if (receiptShowsFailedStatus) {
-            const error: Error = new Error(
-              'Transaction failed. The transaction was dropped or replaced by a new one',
-            );
-            this.failTransaction(meta, error);
-          }
+        if (!txObj && typeof meta.pollingAttempts === 'number') {
```

Comment:
this seems a bit odd to me... is there ever an instance where this property will _not_ be a number?

Reply:
I think the problem will be when the feature gets to prod, the users will have a Transaction object without that property, since it would be introduced in this version, and it could throw an error.

Reply:
in that case I would check for the existence of the prop and _not_ if it's a number eg: `&& 'pollingAttempts' in meta` or something similar.